openmdao/components/tests/test_exec_comp.py

```diff
@@ -1160,6 +1161,40 @@ def test_feature_options(self):
 
         assert_near_equal(prob.get_val('comp.y'), [2., 4.], 0.00001)
 
+    def test_list_outputs_resids_tol(self):
```

Comment:
Do we have a test for residuals_tol when the residual is a scalar?

Reply:
Yes, it's `test_list_residuals_with_tol` in `test_impl_comp.py`

Reply:
If that test works correctly then why didn't it fail when you were missing the residuals_tol check for scalars?

Reply:
Fair point. It was cause I was only checking for an array and not the length of the array